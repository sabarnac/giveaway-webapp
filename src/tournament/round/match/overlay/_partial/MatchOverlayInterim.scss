.match-overlay__interim {
  padding: 4rem;
  display: flex;
  justify-content: space-around;
  align-items: center;
  opacity: 0.999999;
  &--exiting,
  &--exited {
    opacity: 0.000001;
  }
  &--exiting,
  &--entering,
  &--entered {
    opacity: 0.999999;
  }
  & > h3 {
    margin-right: 2 rem;
    text-align: center;
  }
}
