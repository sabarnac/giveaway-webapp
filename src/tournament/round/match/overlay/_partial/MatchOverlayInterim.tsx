import MatchOverlayInterim from "./MatchOverlayInterim.react";
import { withTranslation } from "react-i18next";
import { createObserver } from "../../../../../util";

/**
 * React component for the match overlay interim.
 */
export default withTranslation()(createObserver(MatchOverlayInterim));
