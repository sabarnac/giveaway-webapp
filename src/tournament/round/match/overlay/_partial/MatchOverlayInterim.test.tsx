import React from "react";
import MatchOverlayInterim from "./MatchOverlayInterim";
import { create, ReactTestRenderer } from "react-test-renderer";

describe("MatchOverlayInterim Component.", () => {
  it("Matches snapshot.", () => {
    const component: ReactTestRenderer = create(
      <MatchOverlayInterim
        show={true}
        className="foobar"
        onInterimComplete={() => {}}
      />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
