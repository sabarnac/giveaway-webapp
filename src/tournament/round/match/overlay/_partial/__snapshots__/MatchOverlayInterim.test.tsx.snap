// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MatchOverlayInterim Component. Matches snapshot. 1`] = `
<div
  classNames={
    Object {
      "enter": "",
      "enterActive": "foobar__interim--entering",
      "enterDone": "foobar__interim--entered",
      "exit": "",
      "exitActive": "foobar__interim--exiting",
      "exitDone": "foobar__interim--exited",
    }
  }
  in={false}
  mountOnEnter={true}
  onEntered={[Function]}
  onExited={[Function]}
  timeout={500}
  unmountOnExit={true}
>
  <div
    className="foobar__interim"
    style={
      Object {
        "transition": "opacity 500ms ease-in-out",
      }
    }
  >
    <h3>
      Selecting Winner
    </h3>
    <div
      className="css-yd6m5-style"
    />
  </div>
</div>
`;
